src/main/scala/io/citrine/lolo/trees/Nodes.scala

```diff
@@ -2,30 +2,26 @@ package io.citrine.lolo.trees
 
 import breeze.linalg.DenseMatrix
 import io.citrine.lolo.trees.splits.Split
-import io.citrine.lolo.{Learner, Model, PredictionResult}
-import io.citrine.random.Random
+import io.citrine.lolo.{Model, PredictionResult, TrainingResult}
 
 import scala.collection.mutable
 
 /**
-  * Class to provide getNode interface for internal and leaf training nodes
+  * Trait to provide getNode interface for internal and leaf training nodes
   *
-  * @param trainingData   that this node sees
-  * @param remainingDepth to stop growing the node
-  * @tparam T type of the input vector
-  * @tparam S type of the model output
+  * @tparam T type of the model output
   */
-abstract class TrainingNode[T <: AnyVal, S](
-    trainingData: Seq[(Vector[T], S, Double)],
-    remainingDepth: Int = Int.MaxValue
-) extends Serializable {
+trait TrainingNode[+T] extends Serializable {
```

Comment:
It was unnecessary to have a type parameter for the input and another type parameter for the label, since in practice the inputs are never constrained past `AnyVal`. So now there is only one type parameter, corresponding to the label.